conans/client/cmd/uploader.py

```diff
@@ -185,27 +208,52 @@ def _upload_ref(self, conanfile, ref, prefs, retry, retry_wait, integrity_check,
         # FIXME: because the recipe can have one and the package a different one
         self._hook_manager.execute("pre_upload", conanfile_path=conanfile_path,
                                    reference=ref, remote=recipe_remote)
-
-        self._output.info("Uploading %s to remote '%s'" % (str(ref), recipe_remote.name))
-        self._upload_recipe(ref, conanfile, retry, retry_wait, policy, recipe_remote, remotes)
-        upload_recorder.add_recipe(ref, recipe_remote.name, recipe_remote.url)
-
+        msg = "\rUploading %s to remote '%s'" % (str(ref), recipe_remote.name)
+        self._output.info(left_justify_message(msg))
+        try:
+            self._upload_recipe(ref, conanfile, retry, retry_wait, policy, recipe_remote, remotes)
+            upload_recorder.add_recipe(ref, recipe_remote.name, recipe_remote.url)
+        except ConanException as exc:
+            self._exceptions_list.append(exc)
```

Comment:
If there is an exception uploading the recipe, stop, do not try to upload binaries, you can end this thread now.